tests/unit/display.t

```diff
@@ -150,4 +151,59 @@ $product_ref = {
 $expected = lang('to_do_status') . separator_before_colon($lc) . q{:};
 like(display_field($product_ref, 'states'), qr/$expected/);
 
+# should not loose the second facet at the end of the url on redirection
+my $facets_ref = {
+	'current_link' => 'category/en:lemonades/data-quality',
+	'groupby_tagtype' => 'data_quality',
+	'tagid' => 'en:lemonade'
+};
+
+my $apache_util_module = Test::MockModule->new('Apache2::RequestUtil');
+$apache_util_module->mock(
+	'request',
+	sub {
+		# Return a mock Apache request object
+		my $r = {};
+		bless $r, 'Apache2::RequestRec';
+
+		return $r;
+	}
+);
+
+my $request_rec_module = Test::MockModule->new('Apache2::RequestRec');
+$request_rec_module->mock(
+	'rflush',
+	sub {
+		# Do nothing, am just mocking the method
+	}
+);
+
+$request_rec_module->mock(
+	'status',
+	sub {
+		# Do nothing, am just mocking the method
+	}
+);
+
+$request_rec_module->mock(
+	'headers_out',
+	sub {
+		# Do nothing, am just mocking the method
+
+	}
+);
+
+my $display_module = Test::MockModule->new('ProductOpener::Display');
+$display_module->mock(
+	'redirect_to_url',
+	sub {
+		# Do nothing, am just mocking the method
+	}
+);
+
+display_tag($facets_ref);
+
+unlike($facets_ref->{'current_link'}, qr/en:lemonades\/data\-quality/);    # verify that link has been processed
```

Comment:
instead of like() , unlike(), I think it's best to use is() in order to check the exact value and verify it's the one we expect

e.g. is($facets_ref->{'current_link'}, '/category/bread/data-quality')